micro_dl/torch_unet/networks/Unet25D.py

```diff
@@ -0,0 +1,309 @@
+import datetime
+import os
+import time
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from micro_dl.torch_unet.networks.layers.ConvBlock3D import *
+import micro_dl.torch_unet.utils.logging as log
+
+
+class Unet25d(nn.Module):
+    def __name__(self):
+        return "Unet25d"
+
+    def __init__(
+        self,
+        in_channels=1,
+        out_channels=1,
+        in_stack_depth=5,
+        out_stack_depth=1,
+        xy_kernel_size=(3, 3),
+        residual=False,
+        dropout=0.2,
+        num_blocks=4,
+        num_block_layers=2,
+        num_filters=[],
+        task="seg",
+    ):
+        """
+        Instance of 2.5D Unet.
+        1.) https://elifesciences.org/articles/55502
+
+        Architecture takes in stack of 2d inputs given as a 3d tensor and returns a 2d interpretation.
+        Learns 3d information based upon input stack, but speeds up training by compressing 3d
+        information before the decoding path. Uses interruption conv layers in the Unet skip paths to
+        compress information with z-channel convolution.
+
+        :param int in_channels: number of feature channels in (1 or more)
+        :param int out_channels: number of feature channels out (1 or more)
+        :param int input_stack_depth: depth of input stack in z
+        :param int output_stack_depth: depth of output stack
+        :param int/tuple(int, int) xy_kernel_size: size of x and y dimensions of conv kernels in blocks
+        :param bool residual: see name
+        :param float dropout: probability of dropout, between 0 and 0.5
+        :param int num_blocks: number of convolutional blocks on encoder and decoder paths
+        :param int num_block_layers: number of layer sequences repeated per block
+        :param list[int] num_filters: list of filters/feature levels at each conv block depth
+        :param str task: network task (for virtual staining this is regression): 'seg','reg'
+        :param str debug_mode: if true logs features at each step of architecture, must be manually set
+        """
+        super(Unet25d, self).__init__()
+        self.in_channels = in_channels
+        self.num_blocks = num_blocks
+        self.kernel_size = xy_kernel_size
+        self.residual = residual
+        assert (
+            dropout >= 0 and dropout <= 0.5
+        ), f"Dropout {dropout} not in allowed range: [0, 0.5]"
+        self.dropout = dropout
+        self.task = task
+        self.debug_mode = False
+
+        # ----- set static parameters ----- #
+        self.block_padding = "same"
+        down_mode = "avgpool"  # TODO set static avgpool
```

Comment:
down_mode and up_mode refers to pooling type and upsampling type, right? Just a gentle reminder that the more clearly you name your variables, the easier it is to read the code.